AzureMonitor: Appropriately handle trace correlation resources

aangelisc opened the pull request:
This PR ensures resources correlated from a trace operation ID are used and the resulting queries are built as expected.

Part of #63129

aangelisc committed:
Appropriately handle correlation resources

- Retrieve resources in buildQueries step
- Build updated queries based off of correlation results
- Update tests

aangelisc committed:
Fix query building and tests

aangelisc committed:
Fix lint

asimpson commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:
```diff
@@ -543,21 +552,21 @@ func getCorrelationWorkspaces(ctx context.Context, logger log.Logger, query *Azu
 	}
 
-	return query, nil
+	// Remove the first element as that's where the query is run anyway
```
Good comment :heavy_check_mark:

asimpson reviewed (approved):
Those tests, whew :sweat_smile: 

Nice work :trophy:

aangelisc committed:
Ensure multiple resource queries will work

- Correctly set resources to ensure there is no overlapping between multiple resource selection and correlated resources
- Add tests for these specific cases
- Remove unnecessary pointers
- Build queries correctly
